### PR TITLE
sync issue label with github

### PR DESCRIPTION
**Pull request by btkostner**
Implemented #91. Fixes parseAppHubFileIfPossible function. Couldn't help myself from cleaning up schema layout.

### CODE REVIEW

File: app/model/application.js [hunk 1]
```diff
@@ -79,12 +87,16 @@ ApplicationSchema.statics.fetchReleases = function(application) {
...
 
 ApplicationSchema.statics.parseAppHubFileIfPossible = function(application) {
-  return Promise.try(() => {
-    // Parse the .desktop file
-    const appHubFileBuf = new Buffer(application.appHubFileResult.content, 'base64');
-    const appHubData = JSON.parse(appHubFileBuf.toString());
-    application.priceUSD = appHubData.priceUSD;
-    delete application.appHubFileResult;
+  const fullName = application.github.fullName;
+  return gh.request(`GET /repos/${fullName}/contents/.apphub`, {
```
Comment (aroman) [inline app/model/application.js]: ah, was this necessary because of the refactor I did in `dashboard.js` — no more `appHubFileResult`?

You may still want [`Promise.try()`](http://bluebirdjs.com/docs/api/promise.try.html), which rejects the promise if a synchronous error throws (namely `JSON.parse()` throwing if `.apphub` file has malformed JSON). Not sure what should happen exactly, though — if you reject the promise with .try(), the whole request will return a 500 error. Probably not what we want, it would be better to show a nice popover warning message or something. Either way, there should probably be some explicit error handling for that case... not necessary in this PR, but maybe file a bug as something to be worked on later.

File: app/model/application.js [hunk 2]
```diff
@@ -79,12 +87,16 @@ ApplicationSchema.statics.fetchReleases = function(application) {
...
+  })
+  .then(appHubFile => {
+    // Parse the .apphub file
+    const desktopFileBuf = new Buffer(appHubFile.content, 'base64');
+    const appHubFileJSON = JSON.parse(desktopFileBuf.toString())
+    application.priceUSD = appHubFileJSON.priceUSD
+    application.label = appHubFileJSON.issueLabel;
     return application;
   })
   .catch(() => application);
```
Comment (aroman) [inline app/model/application.js]: since you got rid of the .try(), this catch is not needed. in fact, you definitely don't want it here, because if the gh.request promise fails, you want to fail loudly — this function should only be called on repositories that are known to have .apphub files. (might be good to remove the `ifPossible` from the function name to reflect this?)

File: app/model/application.js [hunk 3]
```diff
@@ -185,6 +197,26 @@ ApplicationSchema.statics.syncIssuesToGitHub = function(application) {
...
+    return gh.request(`POST /repos/${fullName}/labels`, {
+      token: application.github.APItoken,
+      body: {
+        name: application.label,
+        color: '3A416F',
+      },
+    })
+  }).then(labelResult => {
+    application.label = labelResult.name;
+    return application
```
Comment (aroman) [inline app/model/application.js]: semicolon

File: app/model/application.js [hunk 4]
```diff
@@ -185,6 +197,26 @@ ApplicationSchema.statics.syncIssuesToGitHub = function(application) {
...
+  })
+  .catch(() => application);
```
Comment (aroman) [inline app/model/application.js]: why is this here?

File: app/model/application.js [hunk 5]
```diff
@@ -185,6 +197,26 @@ ApplicationSchema.statics.syncIssuesToGitHub = function(application) {
     .then(() => application);
 }
 
+ApplicationSchema.statics.syncLabelToGitHub = function(application) {
```
Comment (aroman) [inline app/model/application.js]: it doesn't seem like this is actually called anywhere?

File: app/model/application.js (hunk 4 above)
Comment (btkostner) [inline app/model/application.js]: The function? To sync the Github issue label :)
The catch? Not sure why I put that there.

File: app/model/application.js (hunk 5 above)
Comment (aroman) [inline app/model/application.js]: also — i'm a bit confused about what this function is supposed to return. 

it seems like right now:

If the label exists, it will return a promise contain the result of GET request for the label.

If the label doesn't exist, it will return a promise containing the application data (with the new label set).

If the POST request fails, the function will return a promise containing the application data (with no label set).

Comment (btkostner) [inline app/model/application.js]: It isn't called anywhere. It should be called before an issue is synced, so I guess i'll add the code for that on this pr.

It will always return application. It runs basically the parseAppHubFile and fetchDesktopFile functions, returning an edited, unsaved, application object. I'll add some more logic for in case the POST fails.

File: app/model/application.js (hunk 1 above)
Comment (btkostner) [inline app/model/application.js]: I think it actually makes sense to show explicit errors not only for this function but also the desktop file.
